### PR TITLE
Add `language_awk`

### PR DESCRIPTION
**Pull request by Samueru-sama**
I used the `language_sh` as reference, added the awk.xml symbols from gtkgetsourceview and compared the results with the awk syntax that mousepad uses, that means that some stuff like `-- Ignore numbers that start with dots or slashes` might not be needed I'm not sure.

### CODE REVIEW

**Comment by Samueru-sama** [comment]
@Guldoman Sorry about the delay, I never got the email of the message from 4 days ago.

btw that `"#.*\n"` is something that's also present in `language_sh`

File: manifest.json [hunk 1]
```diff
@@ -670,6 +671,16 @@
       ],
       "version": "0.1"
     },
+    {
+      "description": "Syntax for the [Awk](https://en.wikipedia.org/wiki/AWK)(v1) programming language",
+      "id": "language_autohotkey_v1",
```
[the remark this answers is not in the document]
Comment (Samueru-sama) [reply manifest.json:676]: What exactly? 😅

**Comment by Guldoman** [comment]
> Sorry about the delay, I never got the email of the message from 4 days ago.

Actually I just posted the review. It was "pending", as I made those comments 4 days ago but didn't publish them yet, as I didn't complete the review. So no worries, you couldn't even see the comments yet.

> btw that `"#.*\n"` is something that's also present in `language_sh`

Yeah but I'm not sure it's actually needed. If that's the case, next time someone changes that file I'll ask them to also change that, as it's not really urgent.

File: manifest.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Guldoman) [reply manifest.json:676]: The ID!

File: plugins/language_awk.lua [hunk 2]
```diff
@@ -0,0 +1,83 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Awk script",
+  files = "%.awk$",
+  headers = "^#!.*bin.*awk.*\n",
```
[the remark this answers is not in the document]
Comment (Samueru-sama) [reply plugins/language_awk.lua:7]: I'm not sure about removing the `.*`, maybe the `\n` but not the other.

awk scripts normally start with `#!/usr/bin/env -S awk -f` and don't end in awk. so I'm not sure if I can remove the last `.*`

[the remark this answers is not in the document]
Comment (Guldoman) [reply plugins/language_awk.lua:7]: Both patterns would result in a positive match, so they're equivalent.

File: plugins/language_awk.lua [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    { pattern = "%f[%w_%.%/]%d[%d%.]*%f[^%w_%.]", type = "number"   },
+    -- Operators
+    { pattern = "[!<>|&%[%]:=*]",                 type = "operator" },
+    -- Match parameters
+    { pattern = "%f[%S][%+%-][%w%-_:]+",          type = "function" },
+    { pattern = "%f[%S][%+%-][%w%-_]+%f[=]",      type = "function" },
+    -- Prevent parameters with assignments from been matched as variables
+    {
+      pattern = "%s%-%a[%w_%-]*%s+()%d[%d%.]+",
+      type = { "function", "number" }
```
Comment (Guldoman) [inline plugins/language_awk.lua:29]:
```suggestion
      pattern = "%s%-%a[%w_%-]*()%s+()%d[%d%.]+",
      type = { "function", "normal", "number" }
```

File: plugins/language_awk.lua [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    {
+      pattern = "%s%-%a[%w_%-]*%s+()%a[%a%-_:=]+",
+      type = { "function", "symbol" }
```
Comment (Guldoman) [inline plugins/language_awk.lua:33]:
```suggestion
      pattern = "%s%-%a[%w_%-]*()%s+()%a[%a%-_:=]+",
      type = { "function", "normal", "symbol" }
```

File: plugins/language_awk.lua [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    -- Match variable assignments
+    { pattern = "[_%a][%w_]+%f[%+=]",              type = "keyword2" },
+    -- Match variable expansions
+    { pattern = "${.-}",                           type = "keyword2" },
+    { pattern = "$[%d$%a_@*][%w_]*",               type = "keyword2" },
+    -- Functions
+    { pattern = "[%a_%-][%w_%-]*[%s]*%f[(]",       type = "function" },
```
Comment (Guldoman) [inline plugins/language_awk.lua:41]:
```suggestion
    { pattern = "[%a_%-][%w_%-]*()%s*%f[(]",       type = { "function", "normal" } },
```

File: plugins/language_awk.lua (hunk 3 above)
Comment (Guldoman) [reply plugins/language_awk.lua:29]: What is this matching? Why is there a whitespace match in front?

File: plugins/language_awk.lua (hunk 4 above)
Comment (Guldoman) [reply plugins/language_awk.lua:33]: What is this matching? Why is there a whitespace match in front?

File: plugins/language_awk.lua [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+syntax.add {
+  name = "Awk script",
+  files = "%.awk$",
+  headers = "^#!.*bin.*awk",
+  comment = "#",
+  patterns = {
+    -- $# is a awk special variable and the '#' shouldn't be interpreted
+    -- as a comment.
+    { pattern = "$[%a_@*#][%w_]*",                type = "keyword2" },
```
Comment (Guldoman) [inline plugins/language_awk.lua:12]: Escaping this for clarity, even if not needed.
```suggestion
    { pattern = "%$[%a_@*#][%w_]*",               type = "keyword2" },
```

File: plugins/language_awk.lua [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    {
+      pattern = "%s%-%a[%w_%-]*%s+()%a[%a%-_:=]+",
+      type = { "function", "symbol" }
+    },
+    -- Match variable assignments
+    { pattern = "[_%a][%w_]+%f[%+=]",              type = "keyword2" },
+    -- Match variable expansions
+    { pattern = "${.-}",                           type = "keyword2" },
+    { pattern = "$[%d$%a_@*][%w_]*",               type = "keyword2" },
```
Comment (Guldoman) [inline plugins/language_awk.lua:39]:
```suggestion
    { pattern = "%${.-}",                          type = "keyword2" },
    { pattern = "%$[%d%$%a_@*][%w_]*",             type = "keyword2" },
```

**Comment by Guldoman** [comment]
Should keywords like `BEGIN`, `END`, `NR`, ... be added in the `symbols` table?

**Comment by Samueru-sama** [comment]
> Should keywords like `BEGIN`, `END`, `NR`, ... be added in the `symbols` table?

[I didn't add them because they don't get highlighted in gtksourceview](https://i.imgur.com/PGtDiXn.png)

File: plugins/language_awk.lua (hunk 3 above)
Comment (Samueru-sama) [reply plugins/language_awk.lua:29]: I have no idea, this is from `language_sh`.

Applying the changes doesn't seem to change anything.

**Comment by Guldoman** [comment]
> > Should keywords like `BEGIN`, `END`, `NR`, ... be added in the `symbols` table?
> 
> [I didn't add them because they don't get highlighted in gtksourceview](https://i.imgur.com/PGtDiXn.png)

Yeah but they get highlighted by gedit, GNOME Text Editor, micro, vscode (with awk plugin), ...

**Comment by Samueru-sama** [comment]
> > > Should keywords like `BEGIN`, `END`, `NR`, ... be added in the `symbols` table?
> > 
> > 
> > [I didn't add them because they don't get highlighted in gtksourceview](https://i.imgur.com/PGtDiXn.png)
> 
> Yeah but they get highlighted by gedit, GNOME Text Editor, micro, vscode (with awk plugin), ...

That's odd, one would think that gnome-text-editor would use gtksourceview.

I thought for a second that it was a bug on mousepad, however I just tested it with xed and it also doesn't get highlighted in xed. 

I guess I will add those anyway.

**Comment by Guldoman** [comment]
Looks good, thank you!

There are a few patterns that are probably not needed, but let's keep them in for now, until they cause issues.

**Pull request merged**